redfish-core/lib/assembly.hpp

```diff
@@ -601,6 +614,42 @@ inline void setAssemblyLocationIndicators(
             {
                 doBatteryCM(asyncResp, assembly, readytoremove.value());
             }
+
+            // Special handling for LCD and base panel. This is required to
+            // support concurrent maintenance for base and LCD panel.
+            else if (sdbusplus::message::object_path(assembly).filename() ==
+                         "panel0" ||
+                     sdbusplus::message::object_path(assembly).filename() ==
+                         "panel1")
+            {
+                // Based on the status of readytoremove flag, inventory data
+                // like CCIN and present property needs to be updated for this
+                // FRU.
+                // readytoremove as true implies FRU has been prepared for
+                // removal. Set action as "deleteFRUVPD". This is the api
+                // exposed by vpd-manager to clear CCIN and set present
+                // property as false for the FRU.
+                // readytoremove as false implies FRU has been replaced. Set
+                // action as "CollectFRUVPD". This is the api exposed by
+                // vpd-manager to recollect vpd for a given FRU.
+                std::string action =
```

Comment:
We typically don't have the ternary operator upstream but oh well.

Reply:
std::string action = CollectFRUVPD

if (readytoremove.value())
{
  action = deleteFRUVPD
}

or an if/else would be more common. Again. Oh well.